OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -141,7 +141,10 @@ public AutoTarget(ActorInitializer init, AutoTargetInfo info)
 		void INotifyCreated.Created(Actor self)
 		{
 			conditionManager = self.TraitOrDefault<ConditionManager>();
-			targetPriorities = self.TraitsImplementing<AutoTargetPriority>().ToArray();
+			activeTargetPriorities =
+				self.TraitsImplementing<AutoTargetPriority>()
+				.OrderByDescending(ati => ati.Info.Priority).ToArray()
```

Comment:
Why not .ToArray() on the end? The ToList on :269 could be avoided as well.

Reply:
We can `ToArray` here since the trait and their priorities will not change. But we want the `IsTraitEnabled` to be re-evaluated every time we re-run this query - as that can change over time.

Reply:
👍

Reply:
Can we get a comment to that effect here?